components/brave_extension/extension/brave_extension/content_element_picker.ts

```diff
@@ -264,11 +276,17 @@ const quitElementPicker = () => {
   document.removeEventListener('scroll', elementPickerViewportChanged)
 }
 
-const launchElementPicker = () => {
-  pickerFrame = document.createElement('iframe')
+const attachElementPicker = async () => {
   // "src" is a web accessible resource since the URI is chrome-extension://.
   // This ensures a malicious page cannot modify the iframe contents.
-  pickerFrame.src = chrome.runtime.getURL('elementPicker.html')
+  pickerDiv = document.createElement('div')
+  const shadowRoot = pickerDiv.attachShadow({ mode: 'open' })
+
+  // Will be resolved by webpack to the file content.
+  // It's a trusted content so it's safe to use innerHTML.
+  // eslint-disable-next-line no-unsanitized/property
+  shadowRoot.innerHTML = require('./elementPicker.html?raw')
```

Comment:
@petemill do you have any ideas?
We can just inline the whole HTML here, but it's super ugly.

Reply:
@thypon Do you still believe that's not a good enough solution?
If we don't trust webpack processing there are possible sec issues in almost every browser js component.

Reply:
Well you could use Trusted Types to get a secure html string, but I'm not sure that would actually gain any restrictions here since all it's going to do is wrap it in a function which is basically saying "this is allowed". We'd still be passing in the same string from the build system which we trust.

So I could suggest how we can use the `html_to_wrapper` GN target template to wrap the HTML in a JS function which uses `static_types.ts`'s `static-types` policy and `createHTML` function, but I don't see that it adds anything beneficial here - if someone can think of a reason it would be helpful, I can outline the steps.

Reply:
this is mainly a provencance issue, I guess. We can use trusted types here, I feel.
We can have something like:

```javascript
const requireWithPolicy = trustedTypes.createPolicy("requirePolicy", {
  createHTML: function(requireString) {
    // check if the file is coming from webpack
    // check if the file is read-only
    // check if the permissions are correct for the file
    // TODO: implement the checks

    return require(requireString)
  },
});

let shadowRoot = /* ... */
const escaped = requireWithPolicy.createHTML("./elementPicker.html?raw")
console.log(escaped instanceof TrustedHTML) // true
el.innerHTML = escaped
```

Reply:
1. does it give us any real extra protection?
2. how to implement these checks?
 
 BTW, There is no separate file, it's build-in in the bundle. here the bundle we inject: https://gist.github.com/atuchin-m/8b9c77fdddd29afd58528a1951025c1b

Reply:
Discussed in DM.
The file content is just inlined in the resulting webpack js bundle, it's just a constant.
The only theoretical "weak" point here is `__webpack_require__`, but if it's hijacked, any loaded js module can be replaced, it doesn't create any new attack vectors.